File: Scalas/src/main/scala/Problem2.scala
package problem

import scala.collection.mutable

class Problem2(list: List[Int], exp: List[Int]) extends Problem {
  override def Calculate(): Boolean = {
    var array = Array.fill(list.length){0}

    for ( i <- 0 to list.length-1){

      var total = 1
      for ( j <- 0 to list.length-1){

        if (j != i){
          total = total * list(j)


        }

      }

      array(i) = total
    }

    return true
  }
}

File: Javascript/Problem/Problem/js/Problem2.js
function Problem2(list, exp) {
    this.list = list;
    this.exp = exp;
}

Problem2.prototype.Calculate = function () {
    var res = [this.list.length]
    for (i = 0; i < this.list.length; i++) {

        var total = 1;

        for (j = 0; j < this.list.length; j++) {

            if (j !== i) {
                total = total * this.list[j];
            }

        }
        res[i] = total;
    }

    console.log(res)
    return true;
}